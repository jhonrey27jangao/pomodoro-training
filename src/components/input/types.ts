export type InputProps = {
  type?: "text";
  width?: number;
  height?: "small" | "big";
  placeholder?: string;
  onBlur?: ((event: React.ChangeEvent<HTMLInputElement>) => void) | undefined;
  onChange?: (e: any) => void | undefined;
  value?: number | any;
  onFocus?: () => void;
  id?: string;
  center?: boolean;
  name?: string;
};
