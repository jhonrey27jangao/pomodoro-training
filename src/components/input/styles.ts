import styled from "styled-components";
import { InputProps } from "./types";

export const StyledInput = styled.input<InputProps>`
  width: ${props => props.width}px;
  display: inline;
  font-family: inherit;
  margin-left: 5px;
  border-radius: 5px;
  padding: 5px 5px;
  box-sizing: border-box;
  border: 1px solid #7a7d86;
  -webkit-transition: 0.3s;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  outline: none;
  font-size: 12px;
  &:focus {
    border: 2px solid #03bb85;
  }
`;
