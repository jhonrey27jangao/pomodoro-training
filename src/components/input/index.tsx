import React from "react";
import { StyledInput } from "./styles";
import { InputProps } from "./types";

const InputComponent = React.memo(
  ({ onChange, width, type, placeholder, value }: InputProps) => {
    return (
      <StyledInput
        type={type}
        onChange={onChange}
        width={width}
        placeholder={placeholder}
        value={value}
      />
    );
  }
);

export const Input: React.FC<InputProps> = ({
  onChange,
  width,
  type,
  placeholder,
  value
}) => {
  return (
    <InputComponent
      onChange={onChange}
      width={width}
      type={type}
      placeholder={placeholder}
      value={value}
    />
  );
};
