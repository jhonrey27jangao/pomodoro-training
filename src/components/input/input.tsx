import React from "react";
import { StyledInput } from "./styles";
import { InputProps } from "./types";

const InputComponent = React.memo(
  ({
    onChange,
    width,
    type,
    placeholder,
    value,
    id,
    height,
    center,
    name,
    props,
  }: InputProps) => {
    return (
      <StyledInput
        type={type}
        onChange={onChange}
        width={width}
        placeholder={placeholder}
        value={value}
        id={id}
        center={center}
        height={height}
        name={name}
        props={props}
      />
    );
  }
);

export const Input: React.FC<InputProps> = ({
  onChange,
  width,
  height,
  type,
  placeholder,
  value,
  id,
  center,
  name,
  props,
}) => {
  return (
    <InputComponent
      onChange={onChange}
      width={width}
      height={height}
      type={type}
      placeholder={placeholder}
      value={value}
      id={id}
      center={center}
      name={name}
      props={props}
    />
  );
};
