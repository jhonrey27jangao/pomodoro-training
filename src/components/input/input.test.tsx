import React from "react";
import { render } from "@testing-library/react";
import { Input } from "./input";

describe("Test for Input Component", () => {
  it("should render Input", () => {
    const { container } = render(<Input />);
    expect(container).toMatchSnapshot();
  });

  it("should render Input", () => {
    const { container } = render(<Input width={20} height="big" center />);
    expect(container).toMatchSnapshot();
  });
});
