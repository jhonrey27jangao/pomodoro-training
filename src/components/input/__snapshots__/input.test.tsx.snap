// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Input Component should render Input 1`] = `
<div>
  <input
    class="sc-AxjAm drbfYu"
    value=""
  />
</div>
`;

exports[`Test for Input Component should render Input 2`] = `
<div>
  <input
    class="sc-AxjAm fcaPGr"
    height="big"
    value=""
    width="20"
  />
</div>
`;
